deps/jerry/tools/profiler/j2v8snap.js

```diff
@@ -0,0 +1,189 @@
+    for (var i = 0; i < edges.length; i++) {
+      var edge = edges[i];
+      this.edgesOutput.push (edge.edge_type);
+      var name;
+      name = this.stringMap.get(edge.name);
+      this.edgesOutput.push (name);
+      // v8 node items is a flattern array,
+      // so to_node is to_node_id * node_size
+      this.edgesOutput.push (this.nodeMap.get(edge.to) * 6);
+    }
+  }
+
+  genNodes() {
+    for (var [node_id, index] of this.nodeMap) {
+      var node = this.nodes[index];
+      this.nodesOutput.push (node.node_type);
+      this.nodesOutput.push (this.stringMap.get(node.name));
+      this.nodesOutput.push (node.id);
+      this.nodesOutput.push (node.size);
+      this.nodesOutput.push (this.edgeMap.get(node_id).length);
+      this.nodesOutput.push (0); // trace node id
+
+      this.genNodeEdges(this.edgeMap.get(node_id));
+    }
+  }
+
+  write() {
+    var outputJSON =
+    {
+      'snapshot':
+      {
+        'meta':
+        {
+          'node_fields':
+          [
+            'type',
+            'name',
+            'id',
+            'self_size',
+            'edge_count',
+            'trace_node_id'
+          ],
+          'node_types':
+          [
+            [
+              'hidden',
+              'array',
+              'string',
+              'object',
+              'code',
+              'closure',
+              'regexp',
+              'number',
+              'native',
+              'synthetic',
+              'concatenated string',
+              'sliced string'
+            ],
+            'string',
+            'number',
+            'number',
+            'number',
+            'number',
+            'number'
+          ],
+          'edge_fields':
+          [
+            'type',
+            'name_or_index',
+            'to_node'
+          ],
+          'edge_types':
+          [
+            [
+              'context',
+              'element',
+              'property',
+              'internal',
+              'hidden',
+              'shortcut',
+              'weak'
+            ],
+            'string_or_number',
+            'node'
+          ],
+          'trace_function_info_fields':
+          [
+            'function_id',
+            'name',
+            'script_name',
+            'script_id',
+            'line',
+            'column'
+          ],
+          'trace_node_fields':
+          [
+            'id',
+            'function_info_index',
+            'count',
+            'size',
+            'children'
+          ],
+          'sample_fields':
+          [
+            'timestamp_us',
+            'last_assigned_id'
+          ]
+        },
+        'node_count': this.nodesOutput.length/6,
+        'edge_count': this.edgesOutput.length/3,
+        'trace_function_count': 0
+      },
+      'nodes': this.nodesOutput,
+      'edges': this.edgesOutput,
+      'trace_function_infos': [],
+      'trace_tree': [],
+      'samples': [],
+      'strings': this.strings
+    };
```

Comment:
Could we put this big JSON to another JSON source, and require via `require('./snapshot-tmpl.json')`?